contrib/utilities/print_deprecated.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2017 by the deal.II authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+"""Search the ./include directory and print out a summary of each deprecated
+object, class, function, etc. Compatile with emacs' compilation-mode.
+
+"""
+
+from __future__ import print_function
+
+import datetime
+import errno
+import os
+import subprocess
+import sys
+
+
+def decode_and_split(string):
+    if sys.version_info.major > 2:
+        return str(string, 'utf8').splitlines()
+    else:
+        return unicode(string, errors='replace').splitlines()
+
+
+class DeprecatedDeclaration(object):
+    def __init__(self, line):
+        assert "DEAL_II_DEPRECATED" in line
+        line_contents = line.split(':')
+        self.file_name = line_contents[0]
+        self.line_n = int(line_contents[1])
+
+        n_context_lines = 6
+
+        self.context_lines = list()
+        with open(self.file_name, 'r') as file_handle:
+            for line_n, file_line in enumerate(file_handle, start=1):
+                if 0 <= self.line_n - line_n <= n_context_lines:
+                    self.context_lines.append(file_line)
+
+        git_log_output = subprocess.check_output(["git", "blame", "-p",
```

Comment:
i wonder if one could get from git commit the closest tag in history backwards? Say if something is deprecated right now, we could get to `v8.5.0` -- that's all we need, probably?